src/main/scala/co/datamechanics/delight/DelightStreamingConnector.scala

```diff
@@ -262,7 +263,7 @@ class DelightStreamingConnector(sparkConf: SparkConf) extends Logging {
         publishPayload(
           StreamingPayload(
             dmAppId,
-            firstEvents,
+            firstEvents.map(e => compact(render(JsonProtocolProxy.jsonProtocol.sparkEventToJson(e)))),
```

Comment:
Why is faster if in the end we have to loop over all events to serialize them ? It is not reducing the number of serialization right ?

Reply:
No worries! 
Actually the change is this PR just moves the costly operation of compacting the events in an async thread.

Before the change:
* Spark produces events and puts them in a queue with limited size
* Spark reads the queue and feeds it the Delight agent in a synchronous fashion
* The Delight agent compacts the events and puts them in its own queue
* The Delight agent queue is read by a separate thread which streams the events to the backend

After the change:
* Spark produces events and puts them in a queue with limited size
* Spark reads the queue and feeds it the Delight agent in a synchronous fashion
* The Delight agent puts the events in its own queue, without compacting them
* The Delight agent queue is read by a separate thread which compacts and streams the events to the backend

So basically, Spark does not need to wait for the agent to compact the events and so (hopefully) Spark queue will never reach its size limit.

What's the tradeoff? The Delight agent queue can know grow a lot (before it was the Spark queue). To be tested on super heavy job.